### ISSUE TITLE
Joining a node causes all repl connections to disconnect

### ISSUE THREAD

**Issue by bsparrow435** [basho/riak_repl#460]
In uni-directional replication, when joining a node to an existing cluster, all connections are dosconnected and never re-connected:

```
2013-12-05 16:20:44.775 [info] <0.199.0>@riak_core_gossip:log_membership_changes:372 '143ee-node2@127.0.0.1' joined cluster with status 'joining'
2013-12-05 16:20:44.797 [debug] <0.198.0>@riak_repl:fixup:33 Hooks for repl modes = [{struct,[{<<"mod">>,<<"riak_repl_leader">>},{<<"fun">>,<<"postcommit">>}]},{struct,[{<<"mod">>,<<"riak_repl2_rt">>},{<<"fun">>,<<"postcommit">>}]}]
2013-12-05 16:20:44.798 [info] <0.2074.0>@riak_core_cluster_mgr:become_proxy:713 ClusterManager: '143ee-node1@127.0.0.1' becoming a proxy to '143ee-node2@127.0.0.1'
2013-12-05 16:20:44.798 [debug] <0.2074.0>@riak_core_cluster_mgr:become_proxy:720 ClusterManager: proxy is removing connections to remote clusters:
2013-12-05 16:20:44.798 [debug] <0.2074.0>@riak_core_cluster_conn_sup:remove_remote_connection:43 Disconnecting from remote cluster at: {cluster_by_name,"B"}
2013-12-05 16:20:44.799 [debug] <0.2072.0>@riak_core_connection_mgr:disconnect_from_target:408 Disconnecting from: {cluster_by_name,"B"}
```

This can be reproduced by setting up two 1-node clusters and setting up unidirectional real time. Then, join a node to cluster A and run `riak-repl connections`. This issue does not persist restart but never corrects itself.

**Comment by bsparrow435** [comment]
Follow up with more data:

Joining a node causes that node to **always** become the repl leader. The above log messages about disconnects are from a node already in the cluster. It closes all the connections because the new leader is supposed to re-establish them. Well, the new leader doesn't have the IP to connect so it just sits in a loop of no IPs associated with this cluster, retry.

Calling `riak-repl connect IP:PORT` fixes the issue and also resolves #449.

**Comment by cmeiklejohn** [comment]
Can we confirm this is still a problem given the fix applied in #451 for 1.4.4?

**Comment by cmeiklejohn** [comment]
@bsparrow435 Is this an issue for 1.4 or 2.0?

**Comment by cmeiklejohn** [comment]
Moving to the 1.4.9 milestone.

**Comment by bsparrow435** [comment]
I've confirmed this issue is fixes in 2.0beta1. Closing #460
